src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecorator.java

```diff
@@ -264,6 +266,8 @@ public void setNodeContext(KubernetesNodeContext nodeContext) {
         this.nodeContext = nodeContext;
     }
 
+    private FilePath workspace;
```

Comment:
Do not keep a `FilePath`-valued field. The decorator is `Serializable` (will be stored in `program.dat`), so if the controller resumed in this middle of the `container` step, this would wind up trying to use `FilePathPickle` which is problematic and deprecated.

You could keep a `String`, but do you actually need this information at all? Why would you be copying temporary scripts into the workspace, polluting the sources checkout etc.? Why not just use the system temporary directory?

(Using `WorkspaceList.tempDir` would be fine too, but would again require stashing information across `launch`.)

Reply:
Done, I believe, in https://github.com/jenkinsci/kubernetes-plugin/pull/1724/commits/6302b66d5b68fc8fcc27eef9ceeebc9bd6c71925. I do not fully understand the API here so it may be a better way that avoids the need of storing the CWD by using the `PodContext` or `Node` but I have not found it as I still need a `FilePath` for `Node.workspaceFor`